src/main/java/org/sonar/ucfg/UCFGtoProtobuf.java

```diff
@@ -0,0 +1,158 @@
+      .addAllUcfgEntries(ucfg.entryBlocks().stream().map(BasicBlock::label).map(Label::id).collect(Collectors.toList()))
+      .addAllBasicBlocks(ucfg.basicBlocks().values().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()))
+      .build();
+
+    try {
+      protobufUCFG.writeTo(new FileOutputStream(filename));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private static Ucfg.BasicBlock toProtobuf(BasicBlock basicBlock) {
+    Ucfg.BasicBlock.Builder builder = Ucfg.BasicBlock.newBuilder()
+      .setId(basicBlock.label().id())
+      .setLocation(toProtobuf(basicBlock.locationInFile()))
+      .addAllInstruction(basicBlock.calls().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()));
+
+    if (basicBlock.terminator().type() == Instruction.InstructionType.RET) {
+      Instruction.Ret ret = (Instruction.Ret) basicBlock.terminator();
+      builder.setRet(Ucfg.Return.newBuilder().setLocation(toProtobuf(ret.location())).setReturnedExpression(toProtobuf(ret.getReturnedExpression())));
+    } else {
+      Ucfg.Jump.Builder jump = Ucfg.Jump.newBuilder();
+      Instruction.Jump terminator = (Instruction.Jump) basicBlock.terminator();
+      jump.addAllDestinations(terminator.destinations().stream().map(Label::id).collect(Collectors.toList()));
+      builder.setJump(jump);
+    }
+    return builder.build();
+  }
+
+  private static Ucfg.Instruction toProtobuf(Instruction.AssignCall assignCall) {
+    return Ucfg.Instruction.newBuilder().setLocation(toProtobuf(assignCall.location()))
+      .setLhs(assignCall.getLhs().id())
+      .setMethodId(assignCall.getMethodId())
+      .addAllArgs(assignCall.getArgExpressions().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()))
+      .build();
+  }
+
+  private static Ucfg.Expression toProtobuf(Expression expression) {
+    Ucfg.Expression.Builder builder = Ucfg.Expression.newBuilder();
+    if (expression.isConstant()) {
+      builder.setConst(Ucfg.Constant.newBuilder().setValue(((Expression.Constant) expression).value()).build());
+    } else {
+      builder.setVar(Ucfg.Variable.newBuilder().setName(((Expression.Variable) expression).id()).build());
+    }
+    return builder.build();
+  }
+
+  private static Ucfg.Location toProtobuf(@Nullable LocationInFile locationInFile) {
+    if (locationInFile == null) {
+      return null;
+    }
+    return Ucfg.Location.newBuilder()
+      .setFileId(locationInFile.getFileId())
+      .setStartLine(locationInFile.getStartLine())
+      .setStartLineOffset(locationInFile.getStartLineOffset())
+      .setEndLine(locationInFile.getEndLine())
+      .setEndLineOffset(locationInFile.getEndLineOffset())
+      .build();
+  }
+
+  public static UCFG fromProtobufFile(String filename) {
+    Ucfg.UCFG ucfg = null;
+    try {
+      ucfg = Ucfg.UCFG.parseFrom(new FileInputStream(filename));
+
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Use a logger to log this exception. ([squid:S1148](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1148))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.ucfg%3Asonar-ucfg&pullRequest=1&issues=AWJJDDis5ZbnBykCSQD5&open=AWJJDDis5ZbnBykCSQD5)

[ISSUE_KEY]: AWJJDDis5ZbnBykCSQD5